core/src/main/scala/besom/internal/logging.scala

```diff
@@ -164,13 +173,115 @@ object logging:
       fileName: sourcecode.FileName,
       name: sourcecode.Name,
       line: sourcecode.Line,
-      mdc: MDC[_]
+      mdc: BesomMDC[_] = BesomMDC.empty
     ): Result[Unit] =
       res match
         case Some(r) =>
           r.urn.getValueOrElse(URN.empty).flatMap(urn => log(Level.Error, mdc, urn, streamId, ephemeral, message))
         case None => log(Level.Error, mdc, URN.empty, streamId, ephemeral, message)
 
+  object LocalBesomLogger extends BesomLogger with LoggerSupport[Result[Unit]]:
+    override def close(): Result[Unit]                = Result.unit
+    override def log(record: LogRecord): Result[Unit] = Result(Logger(record.className).log(record))
+    override def log(record: LogRecord, urn: URN, streamId: Int, ephemeral: Boolean): Result[Unit] =
+      log(record)
+
+  class UserLoggerFactory(using ctx: Context):
+    def trace(message: LoggableMessage, res: Option[Resource] = None, streamId: Int = 0, ephemeral: Boolean = false)(
+      using
+      pkg: sourcecode.Pkg,
+      fileName: sourcecode.FileName,
+      name: sourcecode.Name,
+      line: sourcecode.Line,
+      mdc: MDC = MDC.empty
+    ): Output[Unit] = Output {
+      res match
+        case Some(r) =>
+          r.urn
+            .getValueOrElse(URN.empty)
+            .flatMap(urn => ctx.logger.log(Level.Trace, mdc.inner, urn, streamId, ephemeral, message))
+        case None => ctx.logger.log(Level.Trace, mdc.inner, URN.empty, streamId, ephemeral, message)
```

Comment:
nit: Can this be extracted to some separate function? Those function bodies only differ in the log level.